core/src/main/java/cz/o2/proxima/repository/TransformationDescriptor.java

```diff
@@ -79,48 +76,58 @@ Builder addAttributes(Iterable<AttributeDescriptor<?>> attrs) {
       return this;
     }
 
-    Builder disallowTransactions() {
-      this.supportTransactions = false;
+    Builder disableWriteUsingTransactions() {
+      this.writeUsingTransactions = false;
       return this;
     }
 
     TransformationDescriptor build() {
 
       Preconditions.checkArgument(!attrs.isEmpty(), "Please specify at least one attribute");
       Preconditions.checkArgument(transformation != null, "Please specify transformation function");
-      Preconditions.checkArgument(entity != null, "Please specify source entity");
 
       return new TransformationDescriptor(
-          name, entity, attrs, transformation, supportTransactions, filter);
+          name, attrs, transformation, writeUsingTransactions, filter);
     }
   }
 
   @Getter private final String name;
 
-  @Getter private final EntityDescriptor entity;
-
   private final List<AttributeDescriptor<?>> attributes;
 
   @Getter private final Transformation transformation;
 
-  @Getter private final boolean supportTransactions;
+  @Getter private final boolean writeUsingTransactions;
 
   @Getter private final StorageFilter filter;
 
+  @Getter private final boolean transactional;
```

Comment:
Ok, now I'm just curious  why we need `writeUsingTransactions` should not be `transactional` enough?

Reply:
There are two 'transactional' attributes of a transform:
 a) whether its input attributes are defined to have transactions
 b) whether the output written should create a transaction for each write, or not

What we can do is that we create an (in both cases two-valued, so essentially boolean, but might increase readiblity)
```java
enum InputTransactionalMode {
  ENABLED,
  DISABLED
}
enum OutputTransactionalMode {
  ENABLED,
  DISABLED
}
```
and we can use that in the place of `writeUsingTransactions` and `transactional` (which are admittedly not that clear names)